### PR TITLE
Popout: load transcript after delete-and-auto-select (Fixes #178)

### LINKED ISSUE #178: Popout: deleting a conversation highlights the next one but does not load it
## Summary

In popout mode, deleting a conversation from the sidebar visibly selects/highlights the next conversation in the list, but the chat view does not actually load that conversation - it shows the empty "there are no messages" state instead.

## Steps to reproduce

1. Open PersonalAgent and put the chat into popout mode (not the menu-bar popover).
2. In the sidebar, ensure there are at least two conversations.
3. Delete the currently selected conversation via the sidebar action.

## Expected behaviour

After deletion, the next conversation in the sidebar is selected AND its messages are loaded and displayed in the main chat view.

## Actual behaviour

The next conversation in the sidebar is highlighted (visually selected) but the main chat view shows the "no messages" empty state, as if no conversation is active. Clicking on that same highlighted conversation manually does load it correctly, so the data is fine - only the auto-select-after-delete path is broken.

## Hypothesis (for whoever picks this up)

Most likely the delete handler updates the sidebar's selection state (the `is_selected` / highlighted row) but does not emit or invoke the same "load conversation" path that a manual click does. Two likely candidates:

- The deletion path sets the selected id on the sidebar state but skips the presenter call that materialises messages for the new conversation.
- The deletion path emits a "selection changed" event that is not wired to the popout window's chat view (vs. the menu-bar popover's chat view), so the popover would be fine but popout is not.

Worth checking both the menu-bar popover and popout to confirm which path is affected.

## Acceptance criteria

- [ ] In popout mode, deleting a conversation loads the newly-highlighted conversation's messages automatically.
- [ ] If the deleted conversation was the only one, the empty state remains correct.
- [ ] Behaviour is consistent between popout mode and the menu-bar popover.
- [ ] A regression test (unit or UI) covers the delete-and-auto-select flow.

### PR DESCRIPTION
**Pull request by acoliver**
## Summary

Fixes Issue #178: in popout mode, deleting the selected conversation
visibly selects the next conversation in the sidebar, but the main chat
view stays on the "no messages" empty state. Clicking the highlighted
row manually loads the conversation correctly — only the
delete-and-auto-select path is broken.

## Root cause

When the selected conversation is deleted, `reduce_conversation_deleted`
calls `begin_selection_locked(..., BatchNoPublish)`, which updates the
snapshot (sidebar highlight, `ConversationLoadState::Loading`, bumped
`selection_generation`) **but never emits
`UserEvent::SelectConversation`**. The presenter's transcript-loading
path (`ChatPresenter::handle_select_conversation` →
`replay_conversation_messages`) is driven entirely by that user event,
so the chat view is left with no transcript to render. Manual clicks go
through `handle_select_conversation_intent`, which explicitly emits the
event — that's why manual selection works.

## Fix

Thread the pending selection out of the reducer and have the runtime
pump emit the `SelectConversation` event so the existing presenter flow
loads the transcript. No new selection protocol, no duplicate paths.

- Add `BatchReduceResult { changed, pending_selection }` and a new
  `GpuiAppStore::reduce_batch_with_result` that returns it. The
  existing `reduce_batch() -> bool` becomes a thin compatibility
  wrapper so the many test call sites don't need to change.
- Track the auto-selection inside
  `AppStoreInner::pending_selection_event` from
  `reduce_conversation_deleted` (only for the `SelectedDeleted` outcome
  with a successor). The field is cleared on every batch boundary so
  one auto-selection produces exactly one event.
- In `spawn_runtime_bridge_pump`, call `reduce_batch_with_result` and,
  when `pending_selection` is `Some`, emit
  `UserEvent::SelectConversation { id, selection_generation }` — the
  same contract manual clicks use. If transport enqueue fails, fall
  back to recording a `ConversationLoadFailed` for that generation so
  the chat view surfaces the failure instead of silently hanging.

## Tests

New regression tests in `tests/app_store_tests.rs`:

- `deleting_selected_conversation_produces_pending_selection_event` —
  deleting the selected conversation returns
  `pending_selection = Some((successor, loading_generation))` and the
  snapshot enters `Loading` for the successor with the matching
  generation.
- `deleting_non_selected_conversation_produces_no_pending_selection` —
  deleting a non-selected conversation returns `None` and leaves the
  selected conversation's transcript intact.
- `deleting_last_conversation_produces_no_pending_selection` —
  deleting the only conversation returns `None` and lands on
  `ConversationLoadState::Idle`.

Full verification run locally: `cargo fmt --all -- --check`,
`cargo clippy --all-targets -- -D warnings`,
`cargo test --lib --tests`, `cargo xtask guard` — all clean.

## Acceptance criteria

- [x] In popout mode, deleting a conversation loads the newly-highlighted
  conversation's messages automatically.
- [x] If the deleted conversation was the only one, the empty state
  remains correct (`Idle`, no pending event).
- [x] Behaviour is consistent between popout mode and the menu-bar
  popover — the fix lives in the shared store/pump, not in any
  view-specific code.
- [x] Regression tests cover the delete-and-auto-select flow.

Fixes #178

## Summary by CodeRabbit

* **Bug Fixes**
  * Automatic successor conversation selection now reliably triggers loading of the replacement transcript after deleting the selected conversation, preventing the UI from waiting indefinitely.
  * Improved fallback handling when loading the auto-selected conversation fails.

* **Chores**
  * Added regression tests covering conversation-deletion and auto-selection edge cases.

### CODE REVIEW

File: src/ui_gpui/app_store.rs [hunk 1]
```diff
@@ -262,19 +285,41 @@ impl GpuiAppStore {
     ///
     /// Panics if the store mutex is poisoned.
     pub fn reduce_batch(&self, commands: Vec<ViewCommand>) -> bool {
+        self.reduce_batch_with_result(commands).changed
+    }
```
[the remark this answers is not in the document]
Comment (acoliver) [reply src/ui_gpui/app_store.rs:311]: Addressed in 243a63d: added a debug_assert + tracing::warn in the legacy `reduce_batch` wrapper so any future caller that feeds `ConversationDeleted` for the selected conversation through the boolean API will fail loudly in debug and log in release, instead of silently dropping the pending `SelectConversation`. Also migrated the two pre-existing tests that were already exercising that path onto `reduce_batch_with_result`.

**Comment by acoliver** [comment]
Addressed CodeRabbit feedback in 243a63d:

1. **reduce_batch silently dropping pending selection (Minor issue):** Added debug_assert + tracing::warn! in the compat wrapper so any future caller that feeds ConversationDeleted for the selected conversation through it fails loudly. Migrated the two pre-existing tests onto reduce_batch_with_result.
2. **DRY helper for emit + fallback (nitpick):** Extracted emit_select_or_record_failure. Both handle_select_conversation_intent and the pump now call it.
3. **Follow-up tests on pending_selection semantics (nitpick):** Added pending_selection_event_is_one_shot_per_batch and pending_selection_survives_additional_commands_in_same_batch.
